modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/raft/MetaStorageWriteHandler.java

```diff
@@ -234,4 +253,18 @@ private static RevisionCondition.Type toRevisionConditionType(ConditionType type
                 throw new IllegalArgumentException("Unexpected revision condition type " + type);
         }
     }
+
+    void beforeApply(Command command) {
```

Comment:
I'm trying to wrap my head around this method.
First of all, it is executed on each node in the cluster, correct? If it is, then it feels very dangerous, because it may result in different command times at different nodes.
Please explain it thoroughly, I feel like there's a bug here.

Reply:
This method is only executed on the leader, I thought it's clear from the javadoc, but probably not enough as you're not the first who's asking it